extensionsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl

```diff
@@ -72,6 +72,19 @@ namespace Microsoft.Windows.DevHome.SDK
         Windows.Foundation.IAsyncOperation<ProviderOperationResult> CloneRepositoryAsync(IRepository repository, String cloneDestination, IDeveloperId developerId);
     };
 
+    [contract(Microsoft.Windows.DevHome.SDK.DevHomeContract, 2)] interface IRepositoryProvider2
+    requires IRepositoryProvider
+    {
+        String[] GetSearchFieldNames
+        {
+            get;
+        };
+
+        Windows.Foundation.IAsyncOperation<IVector<String> > GetValuesForField(String fieldName, IMapView<String, String> fieldValues, IDeveloperId developerId);
```

Comment:
Should this be a result object?

Reply:
When I wrote the API, I did think about this.  Jon Wiswall and DevTeam folks had a conversation when we added these request objects.  The consensus around the question "Should this be a response object" can be boiled down to this statement.

`If the method needs to communicate failures besides a catastrophic failure, use a response object`

This method is using strings to compute a list of strings.  Nothing much could go wrong.  If something does, an empty list can be returned and the extension can log any errors.  But, I'll bring this up in the review.